census/tests/factory.py

```diff
@@ -27,3 +28,22 @@ def event(**kwargs):
     data.update(kwargs)
 
     return Event(**data)
+
+
+def google_event(**kwargs):
+    # If you see this error when saving the GoogleEvent:
+    #
+    # > django.db.utils.IntegrityError: NOT NULL constraint failed: census_googleevent.event_id
+    #
+    # Create an Event using the event factory, save it, then supply it to the
+    # GoogleEvent factory. Django OneToOne models require that the related
+    # model exists before saving. Maybe there is a better way to handle this.
```

Comment:
This is just a note documenting one of the "gotchas" that I ran into, in case someone uses the factory and runs into the same issue. The test case that uses this factory is correct, it passes in a saved Event as the note describes.

Alternatively, in the factory, we could _force_ you to pass in your own event by not supplying a default and raising an assertion error. However, the note only applies to saving a model. There are plenty of cases for factories creating models as data, _without_ saving them to the database. I think I prefer the code comment :man_shrugging:

Reply:
Is there a reason we don't want to save when creating an instance with the factory? Also, if we want to start really getting into using factories, it might be worth using [Factory Boy](https://factoryboy.readthedocs.io/en/latest/).

Reply:
In my experience, I usually find that there's some cases in tests when you want to create data, but not necessarily as models within the database. For example an Event form, you might want to generate data from an event, but you don't actually want it in the database. Similar if you are testing services, like our google calendar service might not need a model committed to the db.

Reply:
I'm not familiar with Factory Boy, but most factory frameworks differentiate between `.build()`  which creates a data instance and `.create()` which will commit it to the database. Not necessarily with those names.